nasa/earthdata-search
# EDSC-4051: Fix Swodlr issues convert AccessMethod into functional component

# Overview

### What is the feature?

- Turned AccessMethod Component into a functional component
- Broke out the SWODLR form to its own component
- Add the feature for AccessMethod Radio components to be disabled and hold an error message
- Add unit testing for all these features.

### What is the Solution?

Covered in the feature descriptions.

### What areas of the application does this impact?

My Project page and accessMethod manipulation

# Testing

### Reproduction steps

- **Environment for testing:*production*
- **Collection to test with:** C2799438271-POCLOUD
- Granules that work e2e can we found by filtering with the cycle, pass, scene info `*_011_424_027*`


1. Search the Collection referenced above
2. Select some granules (less than 10)
3. Attempt to customize and download
4. All should work as expected
5. Attempt to add greater than 10 granules
6. SWODLR should be disabled.
7. Subtract granules until there are 10 or less again
8. SWODLR should be re-enabled

Regression test:
- ESI order
- ECHO order
- Harmony order
- Variable subset
- Temporal Subset
- Spatial subset

### Attachments

After logging in/having all granules on the collection
![image](https://github.com/user-attachments/assets/b4d88153-53db-4dd4-9f9b-154afc78a8f5)

Too many granules:
![image](https://github.com/user-attachments/assets/1b7bb91e-b223-489d-bca7-b7eedc289047)

Allowed granules (<10)
![image](https://github.com/user-attachments/assets/4248f6cd-566a-4a3a-8022-e490fd692bf7)


# Checklist
